homeassistant/components/input_number/__init__.py

```diff
@@ -47,18 +48,18 @@
 SERVICE_DECREMENT = "decrement"
 
 
-def _cv_input_number(cfg):
+def _cv_input_number(config: dict) -> dict:
     """Configure validation helper for input number (voluptuous)."""
-    minimum = cfg.get(CONF_MIN)
-    maximum = cfg.get(CONF_MAX)
+    minimum = float(config[CONF_MIN])
+    maximum = float(config[CONF_MAX])
```

Comment:
The main schema has already coerced these to a float, we don't need to do it again here. Just silence the warning or cast it.

Reply:
More type safe option is defining `TypedDict` for config which reflects the schema.